packages/launcher-component/src/flows/progress.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { useState } from 'react';
+import { Toolbar, ToolbarGroup, Button } from '@patternfly/react-core';
+import { StatusMessage, LaunchAppPayload } from 'launcher-client';
+
+import { useLauncherClient } from '../contexts/launcher-client-context';
+import { HubNSpoke } from '..';
+import { ProcessingApp } from '../misc/processing-app';
+import { LaunchNextSteps } from '../misc/launch-next-steps';
+
+enum Status {
+  EDITION = 'EDITION', RUNNING = 'RUNNING', COMPLETED = 'COMPLETED', ERROR = 'ERROR'
+}
+
+interface RunState {
+  status: Status;
+  result?: any;
+  error?: any;
+  statusMessages: StatusMessage[];
+}
+
+interface ProgressProps {
+  items: any[];
+  isValid: () => boolean;
+  convert: () => LaunchAppPayload;
```

Comment:
Love your PR!! great refactor, just the right amount of abstraction :-)

As a naming maniac ^^, I would suggest:
- name the `convert` prop `buildAppPayload`
- name the component `LaunchFlow` (instead of `Progress`)